taker-frontend/src/components/Trade.tsx

```diff
@@ -323,6 +341,96 @@ export default function Trade({
                                         </ModalFooter>
                                     </ModalContent>
                                 </Modal>
+
+                                <Modal
```

Comment:
after getting it working, I guess the next step would be extracting this duplicated code and passing the parameters accordingly